gorb/gorb.go

```diff
@@ -0,0 +1,299 @@
+	}
+	var httpClient = &http.Client{
+		Transport: tr,
+		Timeout:   time.Second * 5,
+	}
+
+	return &gorb{
+		serverBaseURL:   serverBaseURL,
+		drainDelay:      drainDelay,
+		instanceIP:      instanceIP,
+		vipLoadbalancer: vipLoadbalancer,
+		backend:         backendDefinition,
+		httpClient:      httpClient,
+	}, nil
+}
+
+type gorb struct {
+	serverBaseURL   string
+	drainDelay      time.Duration
+	instanceIP      string
+	vipLoadbalancer string
+	httpClient      *http.Client
+	backend         []Backend
+}
+
+func (g *gorb) Start() error {
+	return nil
+}
+
+func (g *gorb) ManageLoopBack(action string, arpIgnore int, arpAnnounce int) error {
+
+	var errorArr *multierror.Error
+	vipLoadbalancerArr := strings.Split(g.vipLoadbalancer, ",")
+	for index, vip := range vipLoadbalancerArr {
+		log.Infof(fmt.Sprintf("VIP %s loopback: %s", action, vip))
+		cmd := fmt.Sprintf("ip addr %s %s/32 dev lo label lo:%d", action, vip, index)
+		_, errCmd := exec.Command("bash", "-c", cmd).Output()
+		errorArr = multierror.Append(errorArr, errCmd)
+
+		log.WithFields(log.Fields{"arp_ignore": arpIgnore}).Info("Set back arp_ignore")
+		cmd = fmt.Sprintf("echo %d > /host-ipv4-proc/arp_ignore", arpIgnore)
+		_, errCmd = exec.Command("bash", "-c", cmd).Output()
+		errorArr = multierror.Append(errorArr, errCmd)
+
+		log.WithFields(log.Fields{"arp_announce": arpAnnounce}).Info("Set back arp_announce")
+		cmd = fmt.Sprintf("echo %d > /host-ipv4-proc/arp_announce", arpAnnounce)
+		_, errCmd = exec.Command("bash", "-c", cmd).Output()
+		errorArr = multierror.Append(errorArr, errCmd)
+	}
+
+	return errorArr.ErrorOrNil()
+}
+
+// Stop removes this instance from Gorb
+func (g *gorb) Stop() error {
+
+	var errorArr *multierror.Error
+	for _, backend := range g.backend {
+		backend.BackendConf.Weight = 0
+		err := g.modifyBackend(&backend)
+		if err != nil {
+			log.WithFields(log.Fields{"err": err}).Error("Unable to set the backend weight to 0")
+			errorArr = multierror.Append(errorArr, err)
+		} else {
+			log.WithFields(log.Fields{"backend": backend}).Infof("Set backend weight to 0")
+		}
+	}
+
+	log.Infof("Waiting %v to finish gorb draining", g.drainDelay)
+	time.Sleep(g.drainDelay)
+
+	err := g.ManageLoopBack("del", 0, 0)
+	errorArr = multierror.Append(errorArr, err)
+
+	for _, backend := range g.backend {
+		err := g.removeBackend(&backend)
+		if err != nil {
+			log.WithFields(log.Fields{"err": err}).Error("Unable to remove Backend ")
+			errorArr = multierror.Append(errorArr, err)
+		} else {
+			log.WithFields(log.Fields{"backend": backend}).Infof("Backend succesfully removed")
+		}
+	}
+
+	return errorArr.ErrorOrNil()
+}
+
+func (g *gorb) Health() error {
+	resp, err := g.httpClient.Get(fmt.Sprintf("%s/service", g.serverBaseURL))
+	if err != nil {
+		return fmt.Errorf("Unable to check service details: %v", err)
+	}
+
+	if resp.StatusCode != 200 {
+		return fmt.Errorf("Gorb server is not healthy. Status code: %d", resp.StatusCode)
```

Comment:
Would it be helpful to log the response body here?

Reply:
done